openedx/core/djangoapps/course_apps/handlers.py

```diff
@@ -0,0 +1,49 @@
+"""
+Signal handlers for course apps.
+"""
+from django.dispatch import receiver
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core.djangoapps.course_apps.models import CourseAppStatus
+
+from openedx.core.djangoapps.course_apps.plugins import CourseAppsPluginManager
+from xmodule.modulestore.django import SignalHandler
+
+from .signals import COURSE_APP_STATUS_INIT
+
+
+@receiver(SignalHandler.course_published)
+def update_course_apps(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Whenever the course is published, update the status of course apps in the
+    django models to match their status in the course.
+    """
+    course_apps = CourseAppsPluginManager.get_apps_available_for_course(course_key)
+    for course_app in course_apps:
+        is_enabled = course_app.is_enabled(course_key=course_key)
+        CourseAppStatus.objects.update_or_create(
```

Comment:
I can see the this code here again -- Can we move this _update_ logic inside a class method in data model?